### PR TITLE
Add GitOps Operator for AppSet deployment

### PR DESCRIPTION
**Pull request by prsurve**

### CODE REVIEW

File: ocs_ci/deployment/deployment.py [hunk 1]
```diff
@@ -199,6 +200,91 @@ def do_deploy_submariner(self):
...
+
+            gitops_subscription_manifest = tempfile.NamedTemporaryFile(
+                mode="w+", prefix="gitops_subscription_manifest", delete=False
+            )
+            templating.dump_data_to_temp_yaml(
+                gitops_subscription_yaml_data, gitops_subscription_manifest.name
+            )
+            run_cmd(f"oc create -f {gitops_subscription_manifest.name}")
+            logger.info("Sleeping for 90 seconds after subscribing to GitOps")
+            time.sleep(90)
```
[the remark this answers is not in the document]
Comment (prsurve) [reply ocs_ci/deployment/deployment.py:238]: we need sleep to git exact csv

**Comment by prsurve** [comment]
verified locally https://url.corp.redhat.com/7fc7701

File: ocs_ci/deployment/deployment.py [hunk 2]
```diff
@@ -199,6 +200,102 @@ def do_deploy_submariner(self):
...
+        Returns:
+
+        """
+        if not config.ENV_DATA.get("deploy_gitops_operator"):
+            return
+
+        # Multicluster operations
+        if config.multicluster:
+            config.switch_acm_ctx()
+            logger.info("Creating GitOps Operator Subscription")
```
Comment (am-agrawa) [inline ocs_ci/deployment/deployment.py:216]: What does Subscription mean here?

File: ocs_ci/deployment/deployment.py [hunk 3]
```diff
@@ -199,6 +200,102 @@ def do_deploy_submariner(self):
...
+            )
+            package_manifest = PackageManifest(
+                resource_name=constants.GITOPS_OPERATOR_NAME,
+            )
+            gitops_subscription_yaml_data["spec"][
+                "startingCSV"
+            ] = package_manifest.get_current_csv(
+                channel="latest", csv_pattern=constants.GITOPS_OPERATOR_NAME
+            )
+
```
Comment (am-agrawa) [inline ocs_ci/deployment/deployment.py:228]: If possible, pls add a few log messages which helps in understanding/debugging.

File: ocs_ci/deployment/deployment.py (hunk 2 above)
Comment (prsurve) [reply ocs_ci/deployment/deployment.py:216]: Subscription is ocp operator subscription

File: ocs_ci/deployment/deployment.py (hunk 3 above)
Comment (prsurve) [reply ocs_ci/deployment/deployment.py:228]: i don't think we need to add any logs here it just build yaml

**Review by am-agrawa — DISMISSED** [review]

File: ocs_ci/deployment/deployment.py [hunk 4]
```diff
@@ -199,6 +200,102 @@ def do_deploy_submariner(self):
...
+            subscriptions = ocp.OCP(
+                kind=constants.SUBSCRIPTION_WITH_ACM,
+                resource_name=constants.GITOPS_OPERATOR_NAME,
+                namespace=constants.OPENSHIFT_OPERATORS,
+            ).get()
+            gitops_csv_name = subscriptions["status"]["currentCSV"]
+            csv = CSV(
+                resource_name=gitops_csv_name, namespace=constants.GITOPS_NAMESPACE
+            )
+            csv.wait_for_phase("Succeeded", timeout=720)  # 720
```
Comment (keesturam) [inline ocs_ci/deployment/deployment.py:250]: the comment "#720" needs to be removed.

File: ocs_ci/deployment/deployment.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (keesturam) [reply ocs_ci/deployment/deployment.py:240]: can you please add this as a comment for readability

File: ocs_ci/deployment/deployment.py (hunk 4 above)
Comment (prsurve) [reply ocs_ci/deployment/deployment.py]: done

File: ocs_ci/deployment/deployment.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (prsurve) [reply ocs_ci/deployment/deployment.py:241]: Done

**Review by am-agrawa — APPROVED** [review]

**Review by keesturam — APPROVED** [review]

**Pull request merged**